### DISCUSSION TITLE
Access queryKey in mutation's queryFn?

### DISCUSSION THREAD

**Discussion by geekox86** [TanStack/query#4386] [Q&A]
Greetings,

I can access queryKey in queries' queryFn with no problem. But, I am not sure how to do so from within mutations' queryFn.

Plz, help.

**Comment by TkDodo** [comment]
mutations don't support that.

**Reply by geekox86** [reply]
Is there a fundamental reason for this? Maybe I am missing something I am missing about mutations?

**Reply by TkDodo** [reply]
What issue are you trying to solve?

**Reply by geekox86** [reply]
I am trying to feed a mutation function's parameters from its mutation key declaratively, just like queries.

The mutation key is a vue reactive with some computed parts that are used in the parameter feeding.

Because I do not want to pass the parameters imperatively with every call to mutate.

**Reply by TkDodo** [reply]
Mutations are fundamentally different from queries because they run on demand, not automatically.You can call mutate multiple times with different values, too.

so either pass your value to .mutate or to useMutation itself and access it from the closure.
